_pages/dev/security/security-headers.md

```diff
@@ -0,0 +1,139 @@
+
+Since audits likely report the lack of `X-Content-Type-Options`, it is recommended to add this header regardless.
+
+Example:
+
+```
+X-Content-Type-Options: nosniff
+```
+
+**TODO**:
+
+- add `X-Content-Type-Options` for compliance reasons and avoid noisy audits
+
+## Content Security Policy (CSP)
+
+CSP prevents a wide range of attacks, including cross-site scripting and other cross-site injections. The policy can specified in great detail with several _directives_.
+
+The policy should include the following directives:
+
+- the `frame-ancestors` directive to allow the storefront to be loaded in SmartEdit frame.
+- the `script-src` directive to prevent scripts being loaded from unknown locations. This does not only include spartacus static resources, but also third-party JS files for integrations such as SmartEdit, Qualtrics, etc.
+  - The `script-src` directive should not have `unsafe-inline` or `unsafe-eval` to prevent XSS.
+    **Note**: some areas in Spartacus (or 3rd parties) might not work after applying this one.
+- (optional) the `object-src` directive to avoid any insecure execution (i.e. flash, java, etc.).
+- (optional) the `default-src` to limit to use https only
+
+Further restrictions regarding images, fonts or other static resources are not added by default. Customers can extend the policy to meet there security requirements.
+
+The policy must be applied to both SSR rendered pages as well as the static `index.html`.
+
+While most of the policy can be added with a meta tag in the `index.html`, the `frame-ancestors` directive must be ignored when contained in a policy declared via a meta element.
+
+**TODO**:
+
+- Discuss adding CSP in security headers
+  - allow for flexibility
+  - allow for injection of environment URLs
+  - consider using `OCC_BACKEND_BASE_URL_VALUE` to dynamically add the backend API base URL in the policy
+  - distinguish dev, qa and prod
+- Consider a default (templated) policy  
+  Consider a template per environment
+- Consider combining html meta tag driven policy vs headers
+
+Meta tag example
+
+```html
+<meta
+  http-equiv="Content-Security-Policy"
+  content="script-src 'myshop.com'; img-src https://*; child-src 'none';"
+/>
+```
+
+## X-Permitted-Cross-Domain-Policies
+
+The `X-Permitted-Cross-Domain-Policies` is used to grant permissions to handle data cross domains by rich clients such as Adobe Flash or Acrobat.
+
+This is not relevant to the standard Spartacus application, and is therefor not enabled.
+
+## Referrer-Policy
+
+The `Referrer-Policy` HTTP header can be used to avoid any confidential information in the referrer URL.
+
+Spartacus does not add any confidential in public storefront URLs. Having rich referrers is recommended for a public storefront in general. This is why the `Referrer-Policy` is not enabled for Spartacus.
+
+## Expect-CT
+
+**TODO**
+
+- @Michael Rieder – investigate if we need this.
```

Comment:
Hm, tough one.
The certificates that I observe in the CCv2 test environments have the required certificate transparency meta-data in their certificate ("Embedded Signed Certificate Timestamp List" extension). But I'm not sure if that's always the case, e.g. when customers bring their own certificates to CCv2.
On the other hand, certain browsers (like chrome) may already enforce certificate transparency regardless of this header.
I would say, it's not strictly necessary. If customers want to use it, they should check if their certificates or their custom TLS configuration are compatible with it.